thrift/thrift0.13/src/test/java/com/linecorp/armeria/internal/common/thrift/logging/UnmaskingTest.java

```diff
@@ -0,0 +1,352 @@
+import com.linecorp.armeria.common.thrift.logging.ThriftFieldMaskerSelector;
+import com.linecorp.armeria.internal.server.thrift.ThriftVersionDetector;
+
+import testing.thrift.main.FooStruct;
+import testing.thrift.main.InnerFooStruct;
+import testing.thrift.main.NormalFooStruct;
+import testing.thrift.main.OptionalFooStruct;
+import testing.thrift.main.SecretStruct;
+import testing.thrift.main.TypedefedFooStruct;
+
+class UnmaskingTest {
+
+    static Stream<Arguments> jsonProtocolComparison_args() {
+        final Stream.Builder<Arguments> args = Stream.builder();
+        final List<TBase<?, ?>> structs = ImmutableList.of(
+                ThriftMaskingStructs.fooStruct(), new FooStruct(), ThriftMaskingStructs.optionalFooStruct(),
+                new OptionalFooStruct(), ThriftMaskingStructs.requiredFooStruct());
+        for (final TBase<?, ?> struct : structs) {
+            args.add(Arguments.of(struct));
+        }
+        return args.build();
+    }
+
+    @ParameterizedTest
+    @MethodSource("jsonProtocolComparison_args")
+    void deserializer(TBase<?, ?> tBase) throws Exception {
+        final TSerializer tSerializer = new TSerializer(ThriftProtocolFactories.json());
+        final TBaseSelectorCache cache = new TBaseSelectorCache(ImmutableList.of());
+        final String upstreamSer = tSerializer.toString(tBase);
+        final TMaskingDeserializer deserializer =
+                new TMaskingDeserializer(ThriftProtocolFactories.json(), cache);
+        final TBase<?, ?> copied = tBase.deepCopy();
+        copied.clear();
+        deserializer.fromString(copied, upstreamSer);
+        assertThat(copied).isEqualTo(tBase);
+    }
+
+    static Stream<Arguments> typedefJsonProtocolComparison_args() {
+        return Stream.of(Arguments.of(typedefedFooStruct()), Arguments.of(new TypedefedFooStruct()));
+    }
+
+    @ParameterizedTest
+    @MethodSource("typedefJsonProtocolComparison_args")
+    void typedefJsonProtocolComparison(TBase<?, ?> tBase) throws Exception {
+        final TSerializer tSerializer = new TSerializer(ThriftProtocolFactories.json());
+        final TBaseSelectorCache cache = new TBaseSelectorCache(ImmutableList.of());
+        final TMaskingSerializer serializer = new TMaskingSerializer(ThriftProtocolFactories.json(), cache);
+        final String upstreamSer = tSerializer.toString(tBase);
+        final String ser = serializer.toString(tBase);
+        assertThatJson(ser).isEqualTo(upstreamSer);
+    }
+
+    @Test
+    void unmaskTBase() throws Exception {
+        final InnerFooStruct inner = new InnerFooStruct().setStringVal("val1");
+        final SecretStruct struct = new SecretStruct().setInnerFooStruct(inner);
+        final String fieldName = "innerFooStruct";
+        final TBaseSelectorCache cache =
+                new TBaseSelectorCache(ImmutableList.of(ThriftFieldMaskerSelector.of(info -> {
+                    if (fieldName.equals(info.fieldMetaData().fieldName)) {
```

Comment:
Optional) Exposing `fieldName` in the `ThriftFieldInfo` and implementing `toString` in `DefaultThriftFieldInfo` would help users debug Thrift maskers.